cmd/client-stats/main.go

```diff
@@ -0,0 +1,150 @@
+// Package beacon-chain defines the entire runtime of an eth2 beacon node.
+package main
+
+import (
+	"fmt"
+	"os"
+	"time"
+	runtimeDebug "runtime/debug"
+
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+	joonix "github.com/joonix/log"
+	prefixed "github.com/x-cray/logrus-prefixed-formatter"
+
+	"github.com/prysmaticlabs/prysm/cmd/client-stats/flags"
+	"github.com/prysmaticlabs/prysm/shared/clientstats"
+	"github.com/prysmaticlabs/prysm/shared/cmd"
+	"github.com/prysmaticlabs/prysm/shared/journald"
+	"github.com/prysmaticlabs/prysm/shared/logutil"
+	"github.com/prysmaticlabs/prysm/shared/version"
+)
+
+var appFlags = []cli.Flag{
+	cmd.VerbosityFlag,
+	cmd.LogFormat,
+	cmd.LogFileName,
+	cmd.ConfigFileFlag,
+	flags.BeaconnodeMetricsURLFlag,
+	flags.ValidatorMetricsURLFlag,
+	flags.ClientStatsAPIURLFlag,
+	flags.ScrapeIntervalFlag,
+}
+
+func main() {
+	app := cli.App{}
+	app.Name = "client-stats"
+	app.Usage = "daemon to scrape client-stats from prometheus and ship to a remote endpoint"
+	app.Action = run
+	app.Version = version.Version()
+
+	app.Flags = appFlags
+
+	// logging/config setup cargo-culted from beaconchain
+	app.Before = func(ctx *cli.Context) error {
+		// Load flags from config file, if specified.
+		if err := cmd.LoadFlagsFromConfig(ctx, app.Flags); err != nil {
+			return err
+		}
+
+		verbosity := ctx.String(cmd.VerbosityFlag.Name)
+		level, err := logrus.ParseLevel(verbosity)
+		if err != nil {
+			return err
+		}
+		logrus.SetLevel(level)
+
+		format := ctx.String(cmd.LogFormat.Name)
+		switch format {
+		case "text":
+			formatter := new(prefixed.TextFormatter)
+			formatter.TimestampFormat = "2006-01-02 15:04:05"
+			formatter.FullTimestamp = true
+			// If persistent log files are written - we disable the log messages coloring because
+			// the colors are ANSI codes and seen as gibberish in the log files.
+			formatter.DisableColors = ctx.String(cmd.LogFileName.Name) != ""
+			logrus.SetFormatter(formatter)
+		case "fluentd":
+			f := joonix.NewFormatter()
+			if err := joonix.DisableTimestampFormat(f); err != nil {
+				panic(err)
+			}
+			logrus.SetFormatter(f)
+		case "json":
+			logrus.SetFormatter(&logrus.JSONFormatter{})
+		case "journald":
+			if err := journald.Enable(); err != nil {
+				return err
+			}
+		default:
+			return fmt.Errorf("unknown log format %s", format)
+		}
+
+		logFileName := ctx.String(cmd.LogFileName.Name)
+		if logFileName != "" {
+			if err := logutil.ConfigurePersistentLogging(logFileName); err != nil {
+				log.WithError(err).Error("Failed to configuring logging to disk.")
+			}
+		}
+		return nil
+	}
+
+	defer func() {
+		if x := recover(); x != nil {
+			log.Errorf("Runtime panic: %v\n%v", x, string(runtimeDebug.Stack()))
+			panic(x)
+		}
+	}()
+
+	if err := app.Run(os.Args); err != nil {
+		log.Error(err.Error())
+	}
+}
+
+func run(ctx *cli.Context) error {
+	scrapeInterval := 60
```

Comment:
Maybe let's define this as a var at the top of the file, and better to set it as a time.Duration type to reduce ambiguity. I think cliCtx also supports https://pkg.go.dev/github.com/urfave/cli#Context.Duration time.Duration, so a user could specify something such as `1s` or `5m` which is really cool

Reply:
Nice, I like the duration suggestion, changed it